plonky2/src/iop/challenger.rs

```diff
@@ -129,7 +129,7 @@ impl<F: RichField, H: Hasher<F>> Challenger<F, H> {
 
     /// Absorb any buffered inputs. After calling this, the input buffer will be empty, and the
     /// output buffer will be full.
-    fn duplexing(&mut self) {
+    pub fn duplexing(&mut self) {
```

Comment:
I'd suggest we keep this private and do what we need in `state()` (or with the suggestion below it'd be called `compact()`)